src/controllers/action-queue/types/pushPayloadTypes.js

```diff
@@ -14,20 +14,13 @@ export type PriceChangePayload = {
   body: string
 }
 
-const asPriceChangePayload = (message: FirebaseMessagingTypes.RemoteMessage): PriceChangePayload => {
-  const clean = asObject({
-    notification: asObject({
-      body: asString
-    }),
-    data: asObject({
-      type: asValue('price-change'),
-      pluginId: asString
-    })
-  })(message)
-  return { ...clean.notification, ...clean.data }
-}
+const asPriceChangePayloadData = asObject({
+  type: asValue('price-change'),
+  pluginId: asString
+})
 
-export const asPushMessagePayloadToEdgeUri = (data: FirebaseMessagingTypes.RemoteMessage): string => {
-  const { type, ...params } = asPriceChangePayload(data) // extend later with asEither
-  return `edge://${type}?${URL.qs.stringify(params)}`
+export const pushMessagePayloadToEdgeUri = (message: FirebaseMessagingTypes.RemoteMessage): string => {
+  const { type, pluginId } = asPriceChangePayloadData(message.data)
+  const body = asString(message.notification.body)
+  return `edge://${type}?${URL.qs.stringify({ body, pluginId })}`
```

Comment:
Ahh, much more readable! However, I don't think we need the `body` in the query string, since it's a localized string, not something our code can use.

Reply:
The body is displayed in the modal itself as the first sentence